SimpleTiledModel.cs

```diff
@@ -251,32 +258,38 @@ protected override bool Propagate()
 	public override Bitmap Graphics()
 	{
 		Bitmap result = new Bitmap(FMX * tilesize, FMY * tilesize);
+        
+        int[] bmpData = new int[result.Height * result.Width];
 
-		for (int x = 0; x < FMX; x++) for (int y = 0; y < FMY; y++)
+        for (int x = 0; x < FMX; x++) for (int y = 0; y < FMY; y++)
 			{
 				bool[] a = wave[x][y];
 				int amount = (from b in a where b select 1).Sum();
 				double lambda = 1.0 / (from t in Enumerable.Range(0, T) where a[t] select stationary[t]).Sum();
 
 				for (int yt = 0; yt < tilesize; yt++) for (int xt = 0; xt < tilesize; xt++)
 					{
-						if (black && amount == T) result.SetPixel(x * tilesize + xt, y * tilesize + yt, Color.Black);
-						else
-						{
-							double r = 0, g = 0, b = 0;
-							for (int t = 0; t < T; t++) if (wave[x][y][t])
-								{
-									Color c = tiles[t][xt + yt * tilesize];
-									r += (double)c.R * stationary[t] * lambda;
-									g += (double)c.G * stationary[t] * lambda;
-									b += (double)c.B * stationary[t] * lambda;
-								}
-
-							result.SetPixel(x * tilesize + xt, y * tilesize + yt, Color.FromArgb((int)r, (int)g, (int)b));
-						}
+                        if (black && amount == T) bmpData[x * tilesize + xt + (y * tilesize + yt) * FMX * tilesize] = unchecked((int)0xff000000);
```

Comment:
@giawa Yup, I understand the purpose. I just have a thing with stray magic numbers. I don't remember ever coming across a case where I'd prefer them over a simple descriptive constant.

@d-kr I mean, it was was obvious to us and probably to most people, still doesn't mean it will be for everyone. And it's just such a simple fix, that I'd do it without thinking about it twice.